adapters/openweb/openweb.go

```diff
@@ -16,175 +19,104 @@ type adapter struct {
-
-	reqs := make([]*adapters.RequestData, 0, totalReqs)
 
-	imps := request.Imp
-	reqCopy := *request
-	reqCopy.Imp = make([]openrtb2.Imp, totalImps)
-	for _, sourceId := range sourceIds {
-		impIds := sourceIdToImpIds[sourceId]
-		reqCopy.Imp = reqCopy.Imp[:0]
+	return append(requestsToBidder, &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpoint + "?publisher_id=" + org,
+		Body:    requestJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}), nil
+}
 
-		for i := 0; i < len(impIds); i++ {
-			reqCopy.Imp = append(reqCopy.Imp, imps[impIds[i]])
+func extractOrg(request *openrtb2.BidRequest) (string, error) {
+	var err error
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err = json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return "", fmt.Errorf("unmarshal bidderExt: %w", err)
 		}
 
-		body, err := json.Marshal(reqCopy)
-		if err != nil {
-			errors = append(errors, fmt.Errorf("error while encoding bidRequest, err: %s", err))
-			return nil, errors
+		var impExt openrtb_ext.ExtImpOpenWeb
+		if err = json.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+			return "", fmt.Errorf("unmarshal ExtImpOpenWeb: %w", err)
 		}
 
-		reqs = append(reqs, &adapters.RequestData{
-			Method:  "POST",
-			Uri:     a.endpoint + fmt.Sprintf("?aid=%d", sourceId),
-			Body:    body,
-			Headers: headers,
-			ImpIDs:  openrtb_ext.GetImpIDs(reqCopy.Imp),
-		})
+		if impExt.Org != "" {
+			return strings.TrimSpace(impExt.Org), nil
+		}
 	}
 
-	return reqs, errors
-
+	return "", errors.New("no org supplied")
 }
 
-func (a *adapter) MakeBids(bidReq *openrtb2.BidRequest, unused *adapters.RequestData, httpRes *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-
-	if httpRes.StatusCode == http.StatusNoContent {
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
 		return nil, nil
 	}
 
-	if httpRes.StatusCode != http.StatusOK {
-		return nil, []error{&errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Remote server error: %s", httpRes.Body),
-		}}
-	}
-	var bidResp openrtb2.BidResponse
-	if err := json.Unmarshal(httpRes.Body, &bidResp); err != nil {
-		return nil, []error{&errortypes.BadServerResponse{
-			Message: fmt.Sprintf("error while decoding response, err: %s", err),
-		}}
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
 	}
 
-	bidResponse := adapters.NewBidderResponse()
-	var errors []error
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
 
-	for _, sb := range bidResp.SeatBid {
-		for i := 0; i < len(sb.Bid); i++ {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
```

Comment:
`adapters.NewBidderResponseWithBidsCapacity` initializes `bidResponse.Currency` to `USD` 

on line 83 `bidResponse.Currency` is being overridded. This may set `bidResponse.Currency` empty if `response.Cur` is empty

Consider refactoring this as below.
```
bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
if len(response.Cur) != 0 {
  bidResponse.Currency = response.Cur
}
```